File: app/src/main/java/com/qhy040404/libraryonetap/data/NetData.java
package com.qhy040404.libraryonetap.data;

import com.google.gson.Gson;

import java.io.Serializable;

public class NetData implements Serializable {
    Gson gson = new Gson();

    private static class GsonData {
        private String fee;
        private String dynamicRemainFlow;
        private String dynamicUsedFlow;
    }

    public String getFee(String data) {
        GsonData gsonData = gson.fromJson(data, GsonData.class);
        return gsonData.fee;
    }

    public String getDynamicUsedFlow(String data) {
        GsonData gsonData = gson.fromJson(data, GsonData.class);
        return gsonData.dynamicUsedFlow;
    }

    public String getDynamicRemainFlow(String data) {
        GsonData gsonData = gson.fromJson(data, GsonData.class);
        return gsonData.dynamicRemainFlow;
    }
}

File: app/src/main/java/com/qhy040404/libraryonetap/data/NetData.kt
package com.qhy040404.libraryonetap.data

import com.qhy040404.libraryonetap.constant.Constants
import com.qhy040404.libraryonetap.utils.extensions.StringExtension.decode

object NetData {
    fun getFee(data: String) = runCatching {
        data.decode<NetDTO>()?.fee!!
    }.getOrDefault(Constants.GLOBAL_ERROR)

    fun getDynamicUsedFlow(data: String) = runCatching {
        data.decode<NetDTO>()?.dynamicUsedFlow!!
    }.getOrDefault(Constants.GLOBAL_ERROR)

    fun getDynamicRemainFlow(data: String) = runCatching {
        data.decode<NetDTO>()?.dynamicRemainFlow!!
    }.getOrDefault(Constants.GLOBAL_ERROR)
}

data class NetDTO(
    val fee: String? = null,
    val dynamicRemainFlow: String? = null,
    val dynamicUsedFlow: String? = null,
    val flag: String,
)
